cpp/ql/src/semmle/code/cpp/models/interfaces/FlowSource.qll

```diff
@@ -0,0 +1,18 @@
+/**
+ * Provides a class for modeling functions that return data from potentially untrusted sources. To use
+ * this QL library, create a QL class extending `DataFlowFunction` with a
+ * characteristic predicate that selects the function or set of functions you
+ * are modeling. Within that class, override the predicates provided by
+ * `RemoteFlowFunction` to match the flow within that function.
+ */
+
+import cpp
+import FunctionInputsAndOutputs
+import semmle.code.cpp.models.Models
+
+/**
+ * A library function which returns data read from a network connection.
+ */
+abstract class RemoteFlowFunction extends Function {
+  abstract predicate hasFlowSource(FunctionOutput output);
```

Comment:
Would it be any problem to add a `string` parameter to hold the textual description of the flow type, like you did in #3274 to be compatible with the other languages?

Reply:
Done